capstone/src/main/java/com/step902020/capstone/GcsStore.java

```diff
@@ -0,0 +1,74 @@
+package com.step902020.capstone;
+
+import com.google.cloud.storage.*;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+import java.net.URL;
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+
+@Component
+public class GcsStore {
+
+  @Autowired
+  Storage storage;
+
+  /**
+   * generates a form for the user to submit profile picture
+   * @param projectId id of the current project
+   * @param bucketName bucket name preset to spring-bucket-jennysheng
+   * @param blobName name of the object matches user email
+   * @return html form in String
+   */
+  public String generateSignedPostPolicyV4(
+          String projectId, String bucketName, String blobName) {
+
+    PostPolicyV4.PostFieldsV4 fields =
+            PostPolicyV4.PostFieldsV4.newBuilder().AddCustomMetadataField("test", "data").build();
+
+    PostPolicyV4 policy =
+            storage.generateSignedPostPolicyV4(
+                    BlobInfo.newBuilder(bucketName, blobName).build(), 10, TimeUnit.MINUTES, fields);
+
+    // create html form
+    StringBuilder htmlForm =
+            new StringBuilder(
+                    "<form action='"
+                            + policy.getUrl()
+                            + "' method='POST' enctype='multipart/form-data'>\n");
+    htmlForm.append("<label>Profile Image (File needs to be at least 70KB. After clicking submit please refresh the page to see the change):</label>");
+    for (Map.Entry<String, String> entry : policy.getFields().entrySet()) {
+      htmlForm.append(
+              "  <input name='"
+                      + entry.getKey()
+                      + "' value='"
+                      + entry.getValue()
+                      + "' type='hidden' />\n");
+    }
+    htmlForm.append("  <input type='file' name='file' required/><br />\n");
+    htmlForm.append("  <input type='submit' value='Upload File' name='submit' onsubmit='closeImageForm();'/><br />\n");
```

Comment:
The error I had before was caused by the `submit` input field being processed on the backed. You just need to mark it to be ignored by prepending `x-ignore`. I figured this out by looking in the internal Google repository.
After this change, I was able to uploaded my image, which is only 6kb. Frankly, I'm not sure why images over 70Kb have been working for you.
```suggestion
    htmlForm.append("  <input type='submit' value='Upload File' name='x-ignore-submit' onsubmit='closeImageForm();'/><br />\n");
```

Reply:
Instead of committing this change I made the change in another commit.